### PR TITLE
[pvr] cleanup unused variable when loading Providers from DB

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Cleanup unused variable when loading Providers from DB

## Motivation and context

Flagged by cppcheck.

## How has this been tested?

MacOSX, with pvr.vuplus

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/pvr/providers/PVRProviders.cpp [hunk 1]
```diff
@@ -107,14 +107,13 @@ bool CPVRProviders::LoadFromDatabase(const std::vector<std::shared_ptr<CPVRClien
...
   {
-    bool bChanged = false;
-
     CPVRProviders providers;
     database->Get(providers, clients);
 
     for (auto& provider : providers.GetProvidersList())
     {
-      bChanged |= (CheckAndAddEntry(provider, ProviderUpdateMode::BY_DATABASE) != nullptr);
+      if (provider)
```
Comment (ksooo) [inline xbmc/pvr/providers/PVRProviders.cpp:115]: Why this check? Can it actually happen that provider is nullptr?

Comment (phunkyfish) [reply xbmc/pvr/providers/PVRProviders.cpp:115]: By the definition of `CheckAndAddEntry()`, yes it’s possible. In this particular case I don’t think so but likely easier reading to stick to the function return values.

Comment (ksooo) [reply xbmc/pvr/providers/PVRProviders.cpp:115]: Not sure I understand. CheckAndAddEntry can return nullptr, but how could a nullptr-CPVRProvider find its way into the providers list? BTW, the original code does not check that, right?

Comment (phunkyfish) [reply xbmc/pvr/providers/PVRProviders.cpp:115]: Fair point. Will clean that up.

Comment (phunkyfish) [reply xbmc/pvr/providers/PVRProviders.cpp]: Cleanup up.

**Review by ksooo — APPROVED** [review]

**Pull request merged**